addons/api/addon/abilities/session.js

```diff
@@ -14,13 +14,25 @@ export default class SessionAbility extends ModelAbility {
   // =permissions
 
   /**
-   * Only "active" or "pending" sessions may be read.
+   * Only available sessions may be read.
    * @type {boolean}
    */
   get canRead() {
     return (
-      (this.model.isActive || this.model.isPending) &&
+      this.model.isAvailable &&
       (this.hasAuthorizedAction('read:self') || super.canRead)
     );
   }
+
+  /**
+   * Only available sessions may be canceled.
+   * @type {boolean}
+   */
+  get canCancel() {
+    return (
+      this.model.isAvailable &&
+      (this.hasAuthorizedAction('cancel:self') ||
+        this.hasAuthorizedAction('cancel'))
```

Comment:
not a blocker: should this be symmetrical to super.CanRead by introducing `canCancel` in the model abilities?

Reply:
I don't think it can be symmetrical, cancel is a specific permission for sessions which can't be generalized to the base model

Reply:
you are right, forgot only sessions can cancel now

Reply:
none blocking comment: In a comment below, you mention for read, `read` and `read:self` are redundant. If so, shouldn't we just check here for `cancel`? As per my understanding of the comment, that will cover `cancel:self`, isn't?

Reply:
An end user might only have `cancel:self`, when I said it's redundant i mean having _both_ `cancel` and `cancel:self`